security/pkg/server/ca/authenticate/kube_jwt.go

```diff
@@ -49,6 +55,40 @@ type KubeJWTAuthenticator struct {
 
 var _ Authenticator = &KubeJWTAuthenticator{}
 
+type jwtPayload struct {
+	// Aud is JWT token audience - used to identify 3p tokens.
+	// It is empty for the default K8S tokens.
+	Aud []string `json:"aud"`
+}
+
+// isK8SUnbound detects if the token is a K8S unbound token.
+// It is a regular JWT with no audience and expiration, which can
+// be exchanged with bound tokens with audience.
+//
+// This is used to determine if we check audience in the token.
+// Clients should not use unbound tokens except in cases where
+// bound tokens are not possible.
+func isK8SUnbound(jwt string) bool {
```

Comment:
Move this to https://github.com/istio/istio/blob/master/security/pkg/util/jwtutil.go ?